### ISSUE TITLE
Issue Versions Addition to Issue Enhanced

### ISSUE THREAD

**Issue by fivetran-joemarkiewicz** [fivetran/dbt_jira#22]
A customer (@curious-bryan) requested the `jira__issue_enhanced` model to include current enriched fields from the jira `version` table. This feature request would most likely require the following updates (and others possibly):

- Addition of the `version` table within the [dbt_jira_source](https://github.com/fivetran/dbt_jira_source) package.
     - 89% of users have this table
- Modeling to map an issue to it's current version. Jira [docs](https://www.atlassian.com/agile/tutorials/versions#:~:text=Step%201%3A%20Create%20a%20version,for%20example%2C%201.0%20or%202.1.) on versions show that versions are typically set at the project level. However, individual issue versions can be set and are logged within the `issue_multiselect_history` table as `affectsVersions` or `fixVersions` field_ids.

**Comment by curious-bryan** [comment]
**Question:** While considering this feature request, could we also add current `status_category_id` and/or `status_category_name` as another field(s) to `jira__issue_enhanced`?  

**Reason:** Adding either of these columns would help users know whether the current status of a jira issue is either "To Do", "In Progress" or "Done" status category.

**Notes:** 

I believe `status_category_id` is already a field in `jira_source` > `models` > `stg_jira__status`.  So adding `status_category_id` to `jira__issue_enhanced` could be as simple as adding this field to select statement of `jira` > `models` > `intermediate` > `int_jira__issue_join`.  

But adding `status_category_name` may require adding a staging model to `jira_source` to stage the `STATUS_CATEGORY` table loaded from the Fivetran connector then joining to that table at the appropriate point.  

However, if the rows in `STATUS_CATEGORY` are immutable jira (not sure), then perhaps a simple case statement could be applied to `jira_source` > `models` > `stg_jira__status` or at another appropriate point.   For example, this simple case statement may be appropriate if status category IDs are immutably linked to status category names in jira:

```
  case
    when status_category_id = 2
    then 'To Do'
    when status_category_id = 3
    then 'Done'
    when status_category_id = 4
    then 'In Progress'
  end
```

**Comment by fivetran-jamie** [comment]
@curious-bryan are there particular columns from the `versions` table you'd like to see pulled in? since an issue can be tagged with multiple versions, we will probably aggregate the version names in an `affects_versions` and a `fixes_versions` column in the `issue_enhanced` model. would that info + format suffice for you?

**Comment by curious-bryan** [comment]
@fivetran-jamie, sorry for the late reply.  For our use case, we do not need any other columns from the version table added to the enhanced issue table.  The version name is sufficient.  

SCOPE CREEP ALERT!  Having said that, it would be clutch to have an intermediate table that matches issue ids to version ids with a date field of when the issue was added to that version.  The grain would be one issue-version pair per row.  That way it would supes easy to later match issues to version history.  If this is an interesting story for dbt_jira, let me know I can create a new github issue.

**Comment by fivetran-jamie** [comment]
@curious-bryan ah yeah that is interesting (and would fit pretty naturally in the model flow) -- could you create a new issue with any other details you'd like to add?

**Comment by curious-bryan** [comment]
Thanks for the encouragement, @fivetran-jamie!  I added a feature request in Issue https://github.com/fivetran/dbt_jira/issues/33.

**Comment by fivetran-jamie** [comment]
included in v0.4.0!
